netdata-agent-deployment/ansible-xsrv/README.md

```diff
@@ -0,0 +1,252 @@
+openssl rand -base64 30 > ~/myproject/.ansible-vault-password
+```
+
+Inside the project directory, create the following files:
+
+**~/myproject/ansible.cfg**
+
+```ini
+[defaults]
+# default inventory file
+inventory = inventory.yml
+# pretty-print ansible output (see ansible-doc -t callback -l for available stdout_callbacks)
+stdout_callback = ansible.posix.debug
+# colon-separated paths in which Ansible will search for collections content
+collections_paths = ./
+# ansible-vault password file to use
+vault_password_file = .ansible-vault-password
+```
+
+**~/myproject/inventory.yml**
+
+```yaml
+all:
+  hosts:
+    my.CHANGEME.org:
+```
+
+**~/myproject/playbook.yml**
+
+```yaml
+- hosts: my.CHANGEME.org
+  roles:
+    - nodiscc.xsrv.monitoring_netdata
+```
+
+**~/myproject/host_vars/my.CHANGEME.org/my.CHANGEME.org.yml**
+
+```yaml
+# SSH host/port, if different from my.CHANGEME.org/22
+# ansible_host: "1.2.3.4"
+# ansible_port: 22
+```
+
+**~/myproject/host_vars/my.CHANGEME.org/my.CHANGEME.org.vault.yml**
+
+```yaml
+# administrator (sudo) account username/password
+ansible_user: "deploy"
+ansible_become_pass: "CHANGEME"
+```
+
+**~/myproject/requirements.yml**
+
+```yaml
+collections:
+  - name: https://gitlab.com/nodiscc/xsrv.git
+    type: git
+    version: release
+```
+
+Install the collection:
+
+```bash
+cd ~/myproject/
+ansible-galaxy collection install --force -r requirements.yml
+```
+
+Encrypt the "vault" file containing sensitive configuration values:
+
+```bash
+ansible-vault encrypt host_vars/my.CHANGEME.org/my.CHANGEME.org.vault.yml
+```
+
+-------------------
+
+
+If needed, add any changes to the default configuration, to your host's `host_vars` files. For example:
+
+```yaml
+# host_vars/my.CHANGEME.org.vault.yml
+# SSH host/port, if different from my.CHANGEME.org/22
+# ansible_host: "1.2.3.4"
+# ansible_port: 22
+
+# don't setup additional/custom netdata modules
+setup_netdata_debsecan: no
+setup_netdata_logcount: no
+setup_needrestart: no
+# set netdata charts update frequency
+netdata_update_every: 5
+# example http check
+netdata_http_checks:
+  - name: example.com
+    url: https://example.com
+    status_accepted:
+      - 200
+      - 401
+    timeout: 10
+    update_every: 20
+```
+
+See the [role defaults](https://github.com/nodiscc/xsrv/blob/master/roles/monitoring_netdata/defaults/main.yml) for all available configuration variables.
+
+
+## Deploy
+
+Once host variables are set, deploy the role to the remote host:
+
+```bash
+# activate the virtualenv if not already done
+source .venv/bin/activate
+# (optional) simulate changes (dry-run)
+ansible-playbook --check playbook.yml
+# deploy netdata/apply configuration
+ansible-playbook playbook.yml
+```
+
+## Access netdata
+
+Access your netdata dashboard at https://my.CHANGEME.org:19999
+
+
+## Screencast
+
+[![asciicast](https://asciinema.org/a/bpWXRpyOrBQyJHaFusbM8Slkb.svg)](https://asciinema.org/a/bpWXRpyOrBQyJHaFusbM8Slkb)
```

Comment:
<img width="1337" alt="Screenshot 2022-05-04 at 20 56 47" src="https://user-images.githubusercontent.com/22274335/166750388-f7f1da34-55b4-44cb-bacc-079727e0cbc2.png">

Reply:
https://asciinema.org/a/bpWXRpyOrBQyJHaFusbM8Slkb:

> This recording has been archived
> All unclaimed recordings (the ones not linked to any user account) are automatically archived 7 days after upload.

I did not know that, I will probably redo it when I get some time (I did not keep the source file), I have removed the screencast section for the time being.